crates/api/src/post_report/create.rs

```diff
@@ -40,6 +44,9 @@ pub async fn create_post_report(
   )
   .await?;
 
+  // Don't allow creating reports for removed / deleted posts
```

Comment:
Put this in the function docs of `check_post_deleted_or_removed()` instead of duplicating it in every callsite.

Reply:
And you also need to add the same check when receiving federated reports.

https://github.com/LemmyNet/lemmy/blob/main/crates/apub/src/activities/community/report.rs#L101

Reply:
this is an interesting topic; right now, receiving federated reports for locally resolved content allows admins to review if there may have been federation issues, such as #4118.
if federation was already working well generally, this shouldn't be necessary, but with the issues we're facing today, where "incorrectly" moderating content does not always reliably federate those actions, being able to receive reports for content already federated locally over federation is definitely useful.

Reply:
Still seems like a good check to have. Done now.